crates/telio-firewall/src/firewall.rs

```diff
@@ -501,12 +486,13 @@ impl StatefullFirewall {
     }
 
     fn handle_outbound_icmp<'a>(&self, ip: &impl IpPacket<'a>) {
-        let key = unwrap_option_or_return!(Self::build_icmp_key(ip, false));
         let mut icmp_cache = unwrap_lock_or_return!(self.icmp.lock());
-        // If key already exists, dont change the value, just update timer with entry
-        if let Entry::Vacant(e) = icmp_cache.entry(key) {
-            telio_log_trace!("Inserting new ICMP conntrack entry {:?}", e.key());
-            e.insert(());
+        if let Some(IcmpKey::IcmpConn(key)) = Self::build_icmp_key(ip, false) {
+            // If key already exists, dont change the value, just update timer with entry
+            if let Entry::Vacant(e) = icmp_cache.entry(key) {
+                telio_log_trace!("Inserting new ICMP conntrack entry {:?}", e.key());
+                e.insert(());
+            }
         }
```

Comment:
maybe an else case with a trace log here? Seems to me that failing to build the key and completely skipping processing of this packet (a change to previous behaviour) could be something interesting to see in logs.

Reply:
We are still processing the packet if building the key fail. All code in this file determines if we allow or reject inbound and outbound packets, and we allow all outbound packets, so it's really about determining if we allow or reject inbound packets. If we simplify that, it becomes allow if it's in the whitelist or cache, reject otherwise. This block of code here just determines which ICMP packets end up in the cache, but that doesn't affect whether we allow the outbound packet. 

Adding an else with a log statement would only really be helpful if we are worried about sending malformed ICMP packets. But if we are, we aren't allowing responses to those packets since it's not added to the cache. I can add it if you still think it's useful, but to me it isn't.

Reply:
That kind of log might be useful if it turns out that parsing of packet is subtly wrong and we are failing to parse correct packet. I think it's worth adding trace log for that.

Reply:
I don't think it makes sense to add that log statement here, but I added log statements in a few other places that would be useful for such diagnostic information